pymultimatic/model/timeprogram.py

```diff
@@ -61,9 +62,18 @@ def _validate_start_time(self, attribute: Any, value: Any) -> None:
         if not validator.match(value):
             raise ValueError(value)
 
+    @end_time.validator
+    def _validate_end_time(self, attribute: Any, value: Any) -> None:
+        if not value:
+            return
+
+        validator = re.compile('[0-9]{1,2}:[0-9]{2}')
+        if not validator.match(value):
+            raise ValueError(value)
+
     def __deepcopy__(self, memodict: Any = None) -> 'TimePeriodSetting':
         return TimePeriodSetting(self.start_time, self.target_temperature,
-                                 self.setting)
+                                 self.setting, end_time=self.end_time)
```

Comment:
Since there is now an end_date to take into consideration, `TimeProgram#get_for` and `TimeProgram#get_next` should also be adapted I think